### ISSUE TITLE
[Challenge 0] Gitpod Ports and CORS don't like hardhat chain requests from the application

### ISSUE THREAD

**Issue by jfarid27** [scaffold-eth/scaffold-eth-challenges#41]
# Description

The gitpod environment serves ports at specific urls, instead of on a single served URL. While this usually isn't an issue, because the react application is at a served port 3000, URL requests to a different port URL don't pass a CORs check. This causes breaks in the application when making requests to the localhost chain, making it unusable.

# How to Reproduce

1. Navigate to https://gitpod.id/#https://github.com/scaffold-eth/scaffold-eth-challenges/tree/challenge-0-simple-nft
2. When the application and tasks are fully loaded, open the browser preview at port `3000`. The application should appear.
3. After a few seconds you should see the error seen in the screenshot below.

# Error Information

```
Access to fetch at 'https://8545-amethyst-condor-919k7bwb.ws-us25.gitpod.io/' from origin 'https://3000-amethyst-condor-919k7bwb.ws-us25.gitpod.io' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

<img width="1024" alt="Screen Shot 2022-01-10 at 9 57 52 PM" src="https://user-images.githubusercontent.com/1673206/148874375-f3fd356c-645e-4f9b-a6ed-87aacb04ddf3.png">

# Considerations

- This might be a security hole since the test application now has a public port open. I would recommend allowing it for users that want to get through the challenge, but make it clear its open in the `README`.
- Is anyone really supporting gitpod?

**Comment by codenamejason** [comment]
I believe this was fixed id pr #42 @jfarid27

**Comment by jfarid27** [comment]
This was fixed and should be good. I'll work on other challenge branches in the meantime.
